hudi-utilities/src/main/java/org/apache/hudi/utilities/deltastreamer/HoodieDeltaStreamer.java

```diff
@@ -276,9 +276,19 @@ public static class Config implements Serializable {
             + ". Allows transforming raw source Dataset to a target Dataset (conforming to target schema) before "
             + "writing. Default : Not set. E:g - org.apache.hudi.utilities.transform.SqlQueryBasedTransformer (which "
             + "allows a SQL query templated to be passed as a transformation function). "
-            + "Pass a comma-separated list of subclass names to chain the transformations.")
+            + "Pass a comma-separated list of subclass names to chain the transformations. Transformer can also include "
+            + "a suffix. This suffix can be appended with the property keys to identify properties related to the transformer. "
+            + "E:g - tr1:org.apache.hudi.utilities.transform.SqlQueryBasedTransformer can be used along with property key "
+            + "hoodie.deltastreamer.transformer.sql.tr1. Here tr1 is a suffix used to identify the keys specific to this transformer. "
```

Comment:
can a user have suffix/id in transformer class name and not in the property? lets clarify such behavior in docs in either case and also have checks to guard against misconfiguration.

Reply:
It may help to allow such a scenario though. consider, the following. 

```
tr1:io.bytearray.TransformerA
tr2:io.bytearray.TransformerB
tr3:io.bytearray.TransformerA
```

with configs 

```
io.bytearray.transformera.x=1
io.bytearray.transformera.tr1.y=2
io.bytearray.transformera.tr3.y=3
```
here, we pass in same value for `x` into tr1/tr3. but different value for y? Can we think through this.

Reply:
This scenario is already allowed currently. The behaviour is `configA.idt1` would override any config `configA` defined in properties. If no key corresponding to `configA.idt1` is defined then default is used.
Also currently we do not allow configurations where id is passed for some transformers and not for others:
```
tr1:io.bytearray.TransformerA
io.bytearray.TransformerB
tr3:io.bytearray.TransformerA
```
Please let me know if this looks good.
cc @rmahindra123 @vinothchandar